code/core/arm9/source/Application/GbaDisplayConfigurationService.cpp

```diff
@@ -125,10 +126,17 @@ void GbaDisplayConfigurationService::SetupGbaScreen(const DisplaySettings& displ
 
 void GbaDisplayConfigurationService::SetupColorCorrection(const DisplaySettings& displaySettings)
 {
-    if (displaySettings.gbaColorCorrection == GbaColorCorrection::None)
-    {
-        clut_disableColorCorrection();
-    }
+    const auto getEnum = displaySettings.gbaColorCorrection;
+        const ColorProfile* profile = colorProfileLut[static_cast<size_t>(getEnum)];
+        if (profile)
+            clut_initColorCorrection(profile);
+        else
+            clut_disableColorCorrection();
+}
+
+void GbaDisplayConfigurationService::SetupDisplayGamma(const DisplaySettings& displaySettings)
```

Comment:
This can be removed when the gamma lut is no longer stateful. Stateful code leads to more bugs and is harder to reuse.